pkgs/characters/lib/src/grapheme_clusters/table.dart

```diff
@@ -1139,10 +1139,10 @@ int low(int codeUnit) {
 @pragma('vm:prefer-inline')
 @pragma('wasm:prefer-inline')
 int high(int lead, int tail) {
-  var offset = (((0x3ff & lead) << 10) + (0x3ff & tail)) + (2048 << 8);
-  var chunkStart = _start.codeUnitAt(offset >> 8);
-  var index = chunkStart + (tail & 255);
-  return _data.codeUnitAt(index);
+  var offset = (tail >> 8) + (lead << 2);
```

Comment:
IIUC, this assumes that `tail` and `lead` don't need to be masked with `0x3ff`. Should this be `assert`ed here?